src/psyclone/parse/module_info.py

```diff
@@ -74,51 +85,58 @@ class ModuleInfo:
     If required it will parse this file and then cache the fparser2 parse tree.
     Similarly, it will also process this parse tree to
     create the corresponding PSyIR which is also then cached.
-
-    :param str name: the module name.
-    :param finfo: object holding information on the source file which defines
-        this module.
-    :type finfo: :py:class:`psyclone.parse.FileInfo`
-
     '''
-    def __init__(self, name, finfo):
-        self._name = name.lower()
-        self._file_info = finfo
-
-        # A cache for the fparser tree
-        self._parse_tree = None
-
-        # Whether we've attempted to parse the source.
-        self._parse_attempted = False
-
-        # A cache for the PSyIR representation
-        self._psyir = None
+    def __init__(
+        self,
+        module_name: str,
+        file_info: FileInfo,
+        psyir_root_node: Node = None,
+    ):
+        """Constructor
+
+        :param module_name: Name of the module
+        :param file_info: FileInfo object to load the module from
+        :param psyir_node: The psyir container as an alternative to
+            loading the psyir from FileInfo, defaults to None
+        """
+        self._name = module_name.lower()
+
+        # Do some checking
+        assert isinstance(file_info, FileInfo)
```

Comment:
You can provide a FileInfo with some dummy information just to provide a FileInfo.
It might be also possible to make it optional.
I actually like this possibility to also set it to None.
I changed this in the constructor and also in the comments.

Reply:
It's now removed.